service/BigText.mo

```diff
@@ -14,6 +14,7 @@ actor {
   public type Id = Nat;
 
   // sequence of text
+
   public type TextSeq = Sequence.Sequence<Text>;
```

Comment:
I guess that answers Joachim's question...

Reply:
Yes, I think that clarifies my terminology a little.

But perhaps someone is (additionally) wondering why a `Text` does not suffice here, since as Joachim points out, a `Text` value is already a kind of sequence?

The answer is that we want to support indexing and query operations whose efficiency we can bound, and I think it's best to use an additional, explicit rope structure in Motoko userland for that purpose.

I call this a `Sequence` in this code, and it can hold generic sequential data (not just textual data), and ensure that the representation is always balanced, which is used by algorithms that read and transform it to ensure their own efficiency (e.g., [lazy sorting by merging subtrees](https://github.com/matthewhammer/motoko-sequence/pull/10)).

Stepping back, I want to use the same sequence structure to hold other sequential data in other projects too, not just in BigText/BigSearch ([more context in this google doc](https://docs.google.com/document/d/1aZBW9YahyhAbro3StTdHTioY277KIgqJGJ-7dV89H7o/edit#heading=h.w1qmlexd6b9y))

I chose this implementation of efficient functional sequences because it isn't complex, I was already familiar with it, and I knew it had the right mathematical properties.